Project-1/Peihong_Xie_Project_1_50375781.md

```diff
@@ -0,0 +1,299 @@
+  ```(a) (¬A→B)∨((A∧¬C)→B) 
+  (b) (A→B)∧(A→¬B)
+  (c) (A→(B∨C))∨(C→¬A) 
+  (d) ((A→B)∧C)∨(A∧D) 
+```
+
+Answer: 
+
+(a) Tautology
+| A | B	| C | ¬A→B | A∧¬C | (A∧¬C)→B | (¬A→B)∨((A∧¬C)→B) |
+| ---  | --- | --- | --- | --- | --- | --- |
+| 1 | 1 | 1 | 1 | 0 | 1 | 1 |
+| 1 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 1 | 1 | 0 | 1 | 1 |
+| 0 | 1 | 1 | 1 | 0 | 1 | 1 |
+| 1 | 0 | 0 | 1 | 1 | 0 | 1 |
+| 0 | 1 | 0 | 1 | 0 | 1 | 1 |
+| 0 | 0 | 1 | 0 | 0 | 1 | 1 |
+| 0 | 0 | 0 | 0 | 0 | 1 | 1 |
+
+(b) Contingent
+| A | B	| A→B | A→¬B | (A→B)∧(A→¬B) | 
+| ---  | --- | --- | --- | --- |
+| 1 | 1 | 1 | 0 | 0 | 
+| 1 | 0 | 0 | 1 | 0 |
+| 0 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 1 | 1 | 1 | 
+
+(c) Tautology
+| A | B	| C | B∨C | A→(B∨C) | C→¬A | (A→(B∨C))∨(C→¬A)) |
+| ---  | --- | --- | --- | --- | --- | --- |
+| 1 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 1 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 1 | 1 | 1 | 0 | 1 |
+| 0 | 1 | 1 | 1 | 1 | 1 | 1 |
+| 1 | 0 | 0 | 0 | 0 | 1 | 1 |
+| 0 | 1 | 0 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 1 | 1 | 1 | 1 | 1 |
+| 0 | 0 | 0 | 0 | 1 | 1 | 1 |
+
+(d) Contingent
+| A | B	| C | D | A→B | (A→B)∧C | A∧D | ((A→B)∧C)∨(A∧D)
+| ---  | --- | --- | --- | --- | --- | --- |--- |
+| 1  | 1 | 1 | 1 | 1 | 1 | 1 | 1 |
+| 1  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 1 | 1 | 0 | 1 | 1 |
+| 1  | 0 | 1 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 1 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 1  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 1 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 1 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 1 | 0 | 0 | 1 | 1 |
+| 0  | 0 | 0 | 1 | 1 | 0 | 0 | 0 |
+| 0  | 0 | 1 | 0 | 1 | 1 | 0 | 1 |
+| 0  | 1 | 0 | 0 | 1 | 0 | 0 | 0 |
+| 1  | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+| 0  | 0 | 0 | 0 | 1 | 0 | 0 | 0 |
+
+[2] A _literal_ is an atomic formula or the negation of an atomic formula. We say a formula is in _conjunctive normal form_ (CNF) if it is the conjunction of the disjunction of literals. Find propositional logic formulas in CNF equivalent to each of the following:
+  ```(a) (A→B)→C
+  (b) (A→(B∨C))∨(C→¬A)
+  (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+```
+Answer:
+
+(a) CNF: (A∨C)∧(¬B∨C)
+
+(b) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
+
+(c) CNF: B∨¬B (the CNF of a tautology is any formula of the form "p∨¬p")
+
+[3] Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+  ```(a)  B(x,y) that says that x is a brother of y
+  (b)  A(x,y) that says that x is an aunt of y
+  (c)  C(x,y) that says that x and y are cousins 
+  (d)  O(x) that says that x is an only child
+  (e)  T(x) that says that x has exactly two brothers 
+```
+
+Answer:
+
+
+(a) B(x,y) $\leftrightarrow$ ∃u(P(u,x)∧P(u,y))∧¬F(x)∧¬(x=y)
+
+(b) A(x,y) $\leftrightarrow$ ∃u∃w(P(u,y)∧P(w,x)∧P(w,u)∧¬(x=u))∧F(x)
```

Comment:
Counterexample: Aunts by marriage

Reply:
> Counterexample: Aunts by marriage

@johnbeve Dear John, I think we should not cosider complexities induced by marriage. After all, we not only have "aunt by marriage, but also have "parent by marriage". Suppose that Jack's biological father died, and his biological mother is married to Duke so Duke becomes Jack's stepfather. So should the possibility of **being stepfather** be covered by P (parent_of)? Also, if we should consider aunts by marriage, then we should also consider brothers by marriage and cousins by marriage, leading to the result that B(x, y) and C (x, y) would also face similar difficulties. I don't think such complexities are necessary.

Reply:
@peihongx I gave you 'parent of' for free, so you're not on the hook for modeling step-parents. 
That said, if your argument is:

1. If we should model aunt by marriage then we should model brother/cousin by marriage
2. We should not model brother/cousin by marriage
3. Hence, we should not model aunt by marriage
 
Then rather than have me accept 3 as you intend, you'll force me to reject 2. Are you sure you want to offer this argument? Maybe a better idea would be to rescind the argument ;)

Reply:
@johnbeve John, I don't think that "aunt by marriage" could be generally defined only by "parent_of". In order to express "aunt by marriage", a core thing is to express "x is a wife of y". Ok, if x and y has a child z, then we can do this by saying that x is a female prarent of z, y is a male parent of z, and x is not equal to y. But if x and y have no child, then it is impossible to express "x is a wife of y" only in terms of "parent_of" because y and y's wife have no relation by blood.